### PR TITLE
LST CPU Optimizations - Trig Simplifications and Redundant Load Removal

### PR DESCRIPTION
**Pull request by GNiendorf**
Speeds up the LST CPU backend, reducing per-event time for LST by roughly 40-50% (excluding pixel seed duplicate cleaning which is disabled in the online version of LST). The changes fall into three categories. First, algebraic trig simplifications replace expensive transcendental calls with equivalent expressions: `sin(atan(x))` becomes `x/sqrt(1+x^2)`, `tan(asin(s))/asin(s)` is computed without calling tan, and `deltaPhi(x1,y1,x2,y2)` which internally calls `atan2` twice is replaced with a single `atan2(cross, dot)` using the 2D cross and dot products directly. Second, loose pre-checks are added that compare the cross product or dot product of hit coordinate vectors against thresholds, effectively cutting on `sin(dPhi)` or `cos(dPhi)` rather than computing the angle first, allowing early rejection of clearly failing candidates before the more expensive exact-angle computation. These are applied in the MiniDoublet barrel and endcap creation kernels, the Segment counting kernel, and the Triplet betaIn check. Third, SoA fields that are invariant across inner loop iterations (module geometry properties, pixel seed kinematics) are pre-loaded once into local structs (ModuleMDData, ModuleSegData, PixelSeedData) to avoid redundant memory reads. GPU timing and physics performance have negligible changes.

There is also a small decrease in the memory footprint from a restructuring of the Segment counting kernel, where we use only the non-trig pre-checks and added an additional cut.

### CODE REVIEW

File: HeterogeneousCore/AlpakaMath/test/alpaka/testDeltaPhi.dev.cc [hunk 1]
```diff
@@ -61,7 +61,12 @@ void testPhiFuncs(uint32_t size, std::vector<double> const& res) {
 
     constexpr T eps = 1.e-5;
     for (size_t i = 0; i < size; ++i) {
-      CHECK_THAT(c_h.data()[i], Catch::Matchers::WithinAbs(res[i], eps));
+      // +/-pi are physically equivalent. Compare absolute value at the boundary
+      if (std::abs(res[i]) == M_PI) {
+        CHECK_THAT(std::abs(c_h.data()[i]), Catch::Matchers::WithinAbs(M_PI, eps));
+      } else {
+        CHECK_THAT(c_h.data()[i], Catch::Matchers::WithinAbs(res[i], eps));
+      }
```
Comment (fwyzard) [inline HeterogeneousCore/AlpakaMath/test/alpaka/testDeltaPhi.dev.cc]: I think this is still quite brittle, because it depends explicitly on the value of `res[i]`.

How about try to support the 2π periodicity with something like
```c++
CHECK_THAT(c_h.data()[i], Catch::Matchers::WithinAbs(res[i], eps) or Catch::Matchers::WithinAbs(res[i] + 2 * M_PI, eps) or Catch::Matchers::WithinAbs(res[i] - 2 * M_PI, eps));
```
?

Of course it could be made more complicated, but this should be enough for all the angles we want to support here.

Comment (GNiendorf) [reply HeterogeneousCore/AlpakaMath/test/alpaka/testDeltaPhi.dev.cc]: I wasn't sure if we wanted to allow the 2pi periodicity in the tests. I'll update it.

Comment (fwyzard) [reply HeterogeneousCore/AlpakaMath/test/alpaka/testDeltaPhi.dev.cc]: Thanks.

I agree that checking explicitly for the corner cases could be preferable, but maybe overkill for these tests.
